bindings/wasm/src/credential/revocation/status_list_2021/entry.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020-2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::WasmStatusPurpose;
+use crate::error::Result;
+use crate::error::WasmResult;
+use identity_iota::core::Url;
+use identity_iota::credential::status_list_2021::StatusList2021Entry;
+use identity_iota::credential::Status;
+use wasm_bindgen::prelude::*;
+
+/// [StatusList2021Entry](https://www.w3.org/TR/2023/WD-vc-status-list-20230427/#statuslist2021entry) implementation.
+#[wasm_bindgen(js_name = StatusList2021Entry, inspectable)]
+pub struct WasmStatusList2021Entry(pub(crate) StatusList2021Entry);
+
+#[wasm_bindgen(js_class = StatusList2021Entry)]
+impl WasmStatusList2021Entry {
+  /// Creates a new {@link StatusList2021Entry}.
+  #[wasm_bindgen(constructor)]
+  pub fn new(
+    status_list: &str,
+    purpose: WasmStatusPurpose,
+    index: usize,
+    id: Option<String>,
+  ) -> Result<WasmStatusList2021Entry> {
+    let status_list = Url::parse(status_list).map_err(|e| JsError::new(&e.to_string()))?;
+    let id = if let Some(id) = id {
+      Some(Url::parse(id).map_err(|e| JsError::new(&e.to_string()))?)
+    } else {
+      None
+    };
+    Ok(Self(StatusList2021Entry::new(status_list, purpose.into(), index, id)))
+  }
+
+  /// Returns this `credentialStatus`'s `id`.
+  #[wasm_bindgen]
+  pub fn id(&self) -> String {
+    self.0.id().to_string()
+  }
+
+  /// Returns the purpose of this entry.
+  #[wasm_bindgen]
+  pub fn purpose(&self) -> WasmStatusPurpose {
```

Comment:
the function signature is compiled to 
```typescript
/**
* Returns the {@link StatusPurpose} of this {@link StatusList2021Credential}.
* @returns {number}
*/
  purpose(): number;
``` 

Which is not correct. There's a limitation of wasm bindgen when it comes to enums. This is why we usually convert Rust's enums to JS classes. For example `WasmMethodScope`. 

It's also generally a good idea to take a look at the file `/node/identity_wasm.d.ts` to check if the compiled output of the bindings matches what you expect.

Reply:
That's weird, I've checked the locally generated `/node/identity_wasm.d.ts` and `StatusPurpose` is always used. I've tried building a couple times always with the same outcome.

Reply:
I updated wasm bindgen on my machine and it seems to work fine, I think they fixed it in one of the latest versions. That's actually pretty nice since it's very useful for what we usually use. Great that we found out about this 👍🏻 You can ignore my comments regarding this.